Feat API add

dkqp opened the pull request:
## What is this PR?🔍
API 기능 점검 및 수정
## Changes or Description📝
delete node API 추가
그 외 API 를 각각 테스트하며 의도대로 동작하지 않는 부분 자잘하게 수정
대략적인 mock 데이터 추가 및 테스트 용 유틸 제작
## Screenshot📸

dkqp committed:
Feat API add

delete node API 추가
그 외 API 를 각각 테스트하며 의도대로 동작하지 않는 부분 자잘하게 수정
대략적인 mock 데이터 추가 및 테스트 용 유틸 제작

forest-im commented on models/MindMap.js:
```diff
@@ -2,10 +2,10 @@ const mongoose = require('mongoose');
-    title: { type: String, required: true },
-    author: { type: String, required: true },
+    title: { type: String },
+    author: { type: mongoose.Schema.Types.ObjectId, required: true },
```
ObjectId로 바꾸셨군요 고생하셨습니다!!

forest-im commented on routes/lib/nodeDeleteHelper.js:
```diff
@@ -0,0 +1,44 @@
+    autopopulate: { maxDepth: 5 },
+  });
+
+  const populatedQueue = [populatedNodes];
```
큐를 사용해서 삭제로직을 작성해주셨군요! autopopulate maxDepth: 5까지 하면 모든 자식 노드를 삭제할 수 있는건가요?

dkqp replied:
데이터를 한번에 가져오기 위해 populate 되는 깊이를 5로 정한 것입니다~
만약 5 깊이 이상으로 자식 노드들이 있다면 재귀를 통해 모두 가져올 때까지 진행합니다.

forest-im reviewed (dismissed):
엄청나게 많이 하셨네요...넘 수고하셨습니다🙇🏻‍♀️

Sharpen-Cjh commented on routes/lib/nodeDeleteHelper.js:
```diff
@@ -0,0 +1,44 @@
+      return nodeDeleteHelper(id);
+    }),
+  );
+
```
Promise.all이 모든 비동기를 실행을 기다려주는 역할인가 보네요
재귀함수까지 이용하셨군요 !
로직구현이 굉장히 어려웠을 것 같은데 고생많으셨습니다.

dkqp replied:
네 맞아요~! Promise.all 을 사용하지 않으면 map 으로 나오는 리스트의 요소가 resolve 되지 않은 채로 나옵니다.

Sharpen-Cjh commented on routes/lib/nodeDeleteHelper.js:
```diff
@@ -0,0 +1,44 @@
+      );
+    }
+
+    deletedIdList.push(tempNode.id);
```
반복문을 통해서 삭제할 ID를 배열안에 모아주는 역할이군요

Sharpen-Cjh commented on spec/MindMapMockData.js:
```diff
@@ -0,0 +1,53 @@
+    headNode: '634e4e47475c008330626939',
+  },
+];
+
```
작업할때 요긴하게 쓰도록 하겠습니다.

dkqp replied:
네, 이 부분은 개선이 필요할 것 같아요 ㅋ

dkqp committed:
Feat API fix

post mindmap 라우팅 및 기능 이슈 해결
app.js 에서 라우팅 중 분실되는 params 를 res.locals 에 저장
User 스키마에 프로필(사진) 추가

Sharpen-Cjh commented on spec/testUtils.js:
```diff
@@ -0,0 +1,70 @@
+  try {
+    await User.insertMany(userData);
+    await MindMap.insertMany(mindMapData);
+    await Node.insertMany(nodeData);
```
데이터를 한번에 넣어주는 메서드도 있었군요!

Sharpen-Cjh reviewed (dismissed):
목데이터도 만드시고 전체적인 코드 정리도 해주셨네요.
고생많으셨습니다!

forest-im commented on models/User.js:
```diff
@@ -3,6 +3,7 @@ const mongoose = require('mongoose');
 const userSchema = new mongoose.Schema({
   userName: { type: String, required: true },
   email: { type: String, required: true, unique: true },
+  profile: { type: String },
```
👍빠르시군용!
